### ISSUE TITLE
Training CLIP-ViT 

### ISSUE THREAD

**Issue by Meteorix** [openai/CLIP#58]
@jongwook Thanks for this great work! 

I am trying to train CLIP VIT B/32 from scratch, but cannot get a higher score on imagenet versus CLIP resnet-50. May I ask what initialization you use in training VIT?

In the paper: ``We closely follow their
implementation with only the minor modification of adding
an additional layer normalization to the combined patch
and position embeddings before the transformer and use a
slightly different initialization scheme.``

**Comment by yuxulingche** [comment]
Did you write the training code yourself？

**Comment by Meteorix** [comment]
> Did you write the training code yourself？

yes, it's not hard.

**Comment by jongwook** [comment]
Are you comparing ViT-based and ResNet50-based models trained with the same dataset? Vision transformers tend to underperform ResNet-based models unless they're trained on a huge dataset, so I'd suspect that could have been the reason, rather than the initialization scheme.

**Comment by Meteorix** [comment]
> Are you comparing ViT-based and ResNet50-based models trained with the same dataset? Vision transformers tend to underperform ResNet-based models unless they're trained on a huge dataset, so I'd suspect that could have been the reason, rather than the initialization scheme.

Thanks for your reply. We have trained on 100M text-image pairs. It turned out that ViTB32 model outperformed ResNet50 model on some of benchmarks.

**Comment by ValerioB88** [comment]
@Meteorix what dataset did you use?

**Comment by dragen1860** [comment]
> @jongwook Thanks for this great work!
> 
> I am trying to train CLIP VIT B/32 from scratch, but cannot get a higher score on imagenet versus CLIP resnet-50. May I ask what initialization you use in training VIT?
> 
> In the paper: `We closely follow their implementation with only the minor modification of adding an additional layer normalization to the combined patch and position embeddings before the transformer and use a slightly different initialization scheme.`

Hi, Meteorix:
  I am also working on this exciting project. I found you have implemented your own training code. That's pretty cool. May i ask for sharing your training code? it's would be good if you can kindly send the code to me. liangqu.long@gmail.com . Thank you very much.

**Comment by jongwook** [comment]
@dragen1860 see #83 and also other third-party implementations by [@KeremTurgutlu](https://github.com/KeremTurgutlu/self_supervised/blob/main/self_supervised/multimodal/clip.py) and [@lucidrains](https://github.com/lucidrains/DALLE-pytorch).

**Comment by JACKHAHA363** [comment]
Hi, I am also running ViT-B/32 and with the open-clip code on CC3M. I also notice a discrepancy between ResNet and ViT. You can see my curves at https://github.com/mlfoundations/open_clip/issues/14. I am interested to know if you solve this performance gap @jongwook @Meteorix

**Comment by Shelton-Zhou** [comment]
> @jongwook Thanks for this great work!
> 
> I am trying to train CLIP VIT B/32 from scratch, but cannot get a higher score on imagenet versus CLIP resnet-50. May I ask what initialization you use in training VIT?
> 
> In the paper: `We closely follow their implementation with only the minor modification of adding an additional layer normalization to the combined patch and position embeddings before the transformer and use a slightly different initialization scheme.`

Hi, Meteorix:
I am also working on this exciting project. I found you have implemented your own training code. May i ask for sharing your training code? it's would be good if you can kindly send the code to me. liangqu.long@gmail.com . Thanks a lot!
